src/System.Runtime/tests/GetTypeCodeTests.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Xunit;
+
+namespace System.Tests
+{
+    public static partial class GetTypeCodeTests
+    {
+        [Theory]
+        [InlineData(TypeCode.Boolean, false)]
+        [InlineData(TypeCode.Char, 'x')]
+        [InlineData(TypeCode.Double, 42.0)]
+        [InlineData(TypeCode.Int32, 42)]
+        [InlineData(TypeCode.Int64, 42L)]
+        [InlineData(TypeCode.Single, 42f)]
+        [InlineData(TypeCode.UInt32, 42u)]
+        [InlineData(TypeCode.UInt64, 42ul)]
+        public static void GetTypeCode_Primitives(TypeCode expected, IConvertible convertible)
+        {
+            Assert.Equal(expected, convertible.GetTypeCode());
+        }
+
+        [Fact]
+        public static void FromByte()
+        {
+            Assert.Equal(TypeCode.Byte, byte.MinValue.GetTypeCode());
+        }
+
+        [Fact]
+        public static void FromDecimal()
+        {
+            Assert.Equal(TypeCode.Decimal, decimal.MinValue.GetTypeCode());
+        }
+
+        [Fact]
+        public static void FromShort()
+        {
+            Assert.Equal(TypeCode.Int16, short.MinValue.GetTypeCode());
+        }
+
+        [Fact]
+        public static void FromSByte()
+        {
+            Assert.Equal(TypeCode.SByte, sbyte.MinValue.GetTypeCode());
+        }
+
+        [Fact]
+        public static void FromUShort()
+        {
+            Assert.Equal(TypeCode.UInt16, ushort.MinValue.GetTypeCode());
+        }
```

Comment:
> Can you explain why Decimal is special?

It's not... that's the point ;)

There is nothing special about Decimal as far as the runtime is concerned.  Primitives like Byte, Char, Int32, etc. are special in that they can be encoded in MSIL metadata.  That's not the case for arbitrary value types, of which Decimal is one as far as the runtime is concerned.  It so happens that the C# language chose to special case Decimal by giving it a language keyword (```decimal```) and a modifier for constructing values (```m```), but it's only special as far as the language is concerned.